frontend/src/employee-frontend/components/unit/tab-calendar/discussion-surveys/survey-editor/DiscussionSurveyForm.tsx

```diff
@@ -0,0 +1,304 @@
+// SPDX-FileCopyrightText: 2017-2024 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { faQuestion } from 'Icons'
+import React, { useCallback, useMemo, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import styled from 'styled-components'
+
+import { useTranslation } from 'employee-frontend/state/i18n'
+import { BoundForm, useFormFields } from 'lib-common/form/hooks'
+import {
+  CalendarEvent,
+  CalendarEventType
+} from 'lib-common/generated/api-types/calendarevent'
+import { cancelMutation } from 'lib-common/query'
+import { UUID } from 'lib-common/types'
+import Button from 'lib-components/atoms/buttons/Button'
+import MutateButton from 'lib-components/atoms/buttons/MutateButton'
+import TreeDropdown, {
+  TreeNode,
+  hasUncheckedChildren
+} from 'lib-components/atoms/dropdowns/TreeDropdown'
+import { TextAreaF } from 'lib-components/atoms/form/TextArea'
+import {
+  FixedSpaceColumn,
+  FixedSpaceRow
+} from 'lib-components/layout/flex-helpers'
+import ExpandingInfo from 'lib-components/molecules/ExpandingInfo'
+import InfoModal from 'lib-components/molecules/modals/InfoModal'
+import { H2, H3, Label } from 'lib-components/typography'
+
+import {
+  createCalendarEventMutation,
+  updateCalendarEventMutation
+} from '../../queries'
+
+import { getPeriodFromDatesOrToday } from './DiscussionTimesForm'
+import { surveyForm } from './form'
+
+const SurveyFormFieldGroup = styled(FixedSpaceColumn).attrs({ spacing: 'S' })``
+const SurveyFormSectionGroup = styled(FixedSpaceColumn).attrs({ spacing: 'L' })`
+  margin-bottom: 60px;
+`
+export const WidthLimiter = styled.div`
+  max-width: 400px;
+`
+export type DiscussionSurveyEditMode = 'create' | 'reserve'
+
+const getTreeSelectionAsRecord = (tree: TreeNode[]) =>
+  Object.fromEntries(
+    tree
+      ?.filter((group) => group.checked)
+      .map((group) => [
+        group.key,
+        hasUncheckedChildren(group)
+          ? group.children
+              ?.filter((child) => child.checked)
+              .map((child) => child.key) ?? []
+          : null
+      ]) ?? []
+  )
+
+export default React.memo(function DiscussionSurveyForm({
+  form,
+  eventData,
+  groupId,
+  unitId
+}: {
+  form: BoundForm<typeof surveyForm>
+  eventData: CalendarEvent | null
+  groupId: UUID
+  unitId: UUID
+}) {
+  const { i18n } = useTranslation()
+  const t = i18n.unit.calendar.events
+
+  const [cancelConfirmModalVisible, setCancelConfirmModalVisible] =
+    useState(false)
+
+  const navigate = useNavigate()
+
+  const { times, title, description, attendees } = useFormFields(form)
+
+  const basicInfoValidationErrors = useMemo(
```

Comment:
Seems like a more correct way to do this might be to wrap the form in `validated()`. Now some of the field level validation comes from the form and some of the validation is added on top ad hoc, which can be confusing and error prone. But there might also be valid reasons why that doesn't work here.